lib/utils/custom-rule-config.js

```diff
@@ -1,11 +1,11 @@
-const fs = require('fs');
-const path = require('path');
+const { existsSync, readFileSync } = require('fs');
+const { resolve } = require('path');
 
 function getCustomRuleConfigFromOptions(options) {
     if (typeof options === 'string') {
-        const customPath = path.resolve(process.cwd(), options);
-        if (fs.existsSync(customPath)) {
-            return JSON.parse(fs.readFileSync(customPath), 'utf8');
+        const customPath = resolve(process.cwd(), options);
+        if (existsSync(customPath)) {
+            return JSON.parse(readFileSync(customPath), 'utf8');
```

Comment:
Skimming the code here and it looks like maybe this happens when linting every file? If that's the case, a bit of caching might help speed things up a bit for folks who use the custom config path option. If you do this, you might want to check the mtime to make it nice on long-lived processes like editor integrations.

Reply:
Absolutely. I thought to store the config in `getCustomRuleConfig`, which should solve that. But most important was to fix the bottleneck you - thankfully! - found. ;-)

Reply:
Yes, thanks for your fast work on that! It looks like we are fully unblocked now. 💖